rust/prql/src/lib.rs

```diff
@@ -57,14 +57,8 @@ pub unsafe extern "C" fn prql_to_sql(
     out: *mut *mut u8,
     out_size: *mut u64,
 ) -> i64 {
-    let ret = panic::catch_unwind(|| {
-        return prql_to_sql_impl(query, size, out, out_size);
-    });
-    return match ret {
-        // NOTE: using cxxbridge we can return proper Result<> type.
-        Err(_err) => 1,
-        Ok(res) => res,
-    }
+    // NOTE: using cxxbridge we can return proper Result<> type.
+    panic::catch_unwind(|| prql_to_sql_impl(query, size, out, out_size)).unwrap_or(1)
```

Comment:
Hm, I've tried your patch, and it did not helps, and also it is unclear to me how this code is different, can you please clarify?

Reply:
You're correct — I'm looking more into this. I'll at least push a test that shows it works on the rust side.

Reply:
OK, the panic was likely because `set_output` was getting called and the panic handler had prevented it from initializing the memory. I could reproduce that in my tests with `x -> y`, though I don't have a test setup for the full stack.

Reply:
Yes, now it works OK, let's also add a test in ClickHouse infrastructure as well, something like this should work.

[0001-Add-a-test-for-PRQL-panics.patch.txt](https://github.com/ClickHouse/ClickHouse/files/14478161/0001-Add-a-test-for-PRQL-panics.patch.txt)

Reply:
Great, thanks for that. I've added it.